app/scripts/modules/amazon/src/loadBalancer/configure/common/LoadBalancerLocation.tsx

```diff
@@ -111,51 +115,102 @@ class LoadBalancerLocationImpl extends React.Component<ILoadBalancerLocationProp
   }
 
   public componentDidMount(): void {
-    if (this.props.isNew || this.props.forPipelineConfig) {
-      this.loadAccounts();
-      const hideInternalFlag = this.shouldHideInternalFlag();
-      this.setState({ hideInternalFlag });
-    }
-
+    this.setState({ hideInternalFlag: this.shouldHideInternalFlag() });
     if (this.props.loadBalancer && this.props.isNew) {
       this.buildName();
     }
-  }
 
-  private loadAccounts(): void {
-    AccountService.listAccounts('aws').then(accounts => {
-      this.setState({ accounts });
-      this.accountUpdated(this.props.formik.values.credentials);
+    const formValues$ = this.props$.map(props => props.formik.values);
+    const selectedAccount$ = formValues$.map(x => x.credentials).distinctUntilChanged();
+    const selectedRegion$ = formValues$.map(x => x.region).distinctUntilChanged();
```

Comment:
I really like that these variables are prefixed with `selected` - when I'm reading the code, it makes it easy to see that these streams are from form values. It might be a nice pattern for the others below (`stack$` -> `selectedStack$`, etc).

Reply:
I moved the observable variables for forms into a separate object in a separate commit.  Do you like this better?
```js
    const form = {
      account$: formValues$.map(x => x.credentials).distinctUntilChanged(),
      region$: formValues$.map(x => x.region).distinctUntilChanged(),
      subnetPurpose$: formValues$.map(x => x.subnetType).distinctUntilChanged(),
      stack$: formValues$.map(x => x.stack).distinctUntilChanged(),
      detail$: formValues$.map(x => x.detail).distinctUntilChanged(),
    };
```

```js
const accountRegions$ = Observable.combineLatest(form.account$, allAccounts$)
      .switchMap(([currentAccount, _allAccounts]) => AccountService.getRegionsForAccount(currentAccount))
      .shareReplay(1);	      .shareReplay(1);
```